### PR TITLE
feat: add analytics

### PR DESCRIPTION
**Pull request by PuneetGopinath**
Status: in progress

### CODE REVIEW

File: src/site/app.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ module.exports.load = (client) => {
         .set("view engine", "ejs")
         // Set the ejs templates to ./views
         .set("views", path.join(__dirname, "/views"))
+        // Get real IPs even if behind proxy
```
Comment (PuneetGopinath) [inline src/site/app.js:25]: @abhijoshi2k This is where I have set the trust proxy

File: src/site/routes/bot.js [hunk 2]
```diff
@@ -15,16 +16,32 @@ router.get("/:botId", async (req, res) => {
...
+        return res.redirect(
+            `/bots/add?error=true&message=${encodeURIComponent(
+                "Bot is not approved so you can't view it.<br>\nIf you are this bot's owner then please login and try again"
+            )}`
+        );
+    if (!req.user || req.user.id !== botDB.owner) {
+        if (isNaN(botDB.analytics.views)) botDB.analytics.views = 0;
+        botDB.analytics.views++;
+        if (process.env.GEOLOC_KEY) {
+            const res = await axios.get(
```
Comment (PuneetGopinath) [inline src/site/routes/bot.js:29]: And here I make request to the api

**Review by peterProgramer — DISMISSED** [review]
no problem found

**Comment by PuneetGopinath** [comment]
@peterProgramer thanks for your suggestion

**Review by neeraj395 — APPROVED** [review]
APPROVED !

**Pull request merged**